src/main/java/com/jfrog/ide/common/persistency/TimeBasedMap.java

```diff
@@ -0,0 +1,35 @@
+package com.jfrog.ide.common.persistency;
+
+import javax.annotation.Nonnull;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * @author yahavi
+ **/
+class TimeBasedMap extends LinkedHashMap<String, ScanCacheObject> {
+
+    /**
+     * We override put in order to avoid inserting expired elements.
+     * @param key - key with which the specified value is to be associated
+     * @param value - value to be associated with the specified key
+     * @return the previous value associated with key, or null if expired or there was no mapping for key.
+     */
+    @Override
+    public ScanCacheObject put(@Nonnull String key, @Nonnull ScanCacheObject value) {
+        if (value.isExpired()) {
+            return null;
+        }
+        return super.put(key, value);
+    }
+
+    /**
+     * Runs after "put" to remove expired entries in cache and free memory.
+     * @param eldest - The least recently inserted entry in the map.
+     * @return true if the eldest entry is expired.
+     */
+    @Override
+    protected boolean removeEldestEntry(Map.Entry<String, ScanCacheObject> eldest) {
```

Comment:
Do we need to define this method?
It looks like we're already doing an "isExpired()" check inside the "put" method.

Reply:
The `isExpired` call in the `put` method makes sure that no old entries are entered into the cache. This is possible when loading cache from the file system.
The `removeEldestEntry` method is called after put in order to delete the oldest entry from the set when inserting a new one. This is an optimization that helps us to reduce memory consumtion.